tests/phpunit/tests/option/wpSetOptionAutoloadValues.php

```diff
@@ -0,0 +1,181 @@
+		foreach ( $options as $option => $autoload ) {
+			$this->assertArrayNotHasKey( $option, wp_cache_get( 'alloptions', 'options' ), sprintf( 'Option %s not deleted from alloptions cache', $option ) );
+		}
+	}
+
+	/**
+	 * Tests setting options' autoload to 'yes' where for all of them this is already the case.
+	 *
+	 * @ticket 58964
+	 */
+	public function test_wp_set_option_autoload_values_all_yes_no_update() {
+		global $wpdb;
+
+		$options = array(
+			'test_option1' => 'yes',
+			'test_option2' => 'yes',
+		);
+		add_option( 'test_option1', 'value1', '', 'yes' );
+		add_option( 'test_option2', 'value2', '', 'yes' );
+		$expected = array(
+			'test_option1' => false,
+			'test_option2' => false,
+		);
+
+		$num_queries = get_num_queries();
+		$this->assertSame( $expected, wp_set_option_autoload_values( $options ), 'Function produced unexpected result' );
+		$this->assertSame( $num_queries + 1, get_num_queries(), 'Function made unexpected amount of database queries' );
+		$this->assertSame( array( 'yes', 'yes' ), $wpdb->get_col( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name IN (" . implode( ',', array_fill( 0, count( $options ), '%s' ) ) . ')', ...array_keys( $options ) ) ), 'Option autoload values not updated in database' );
+		foreach ( $options as $option => $autoload ) {
+			$this->assertArrayHasKey( $option, wp_cache_get( 'alloptions', 'options' ), sprintf( 'Option %s unexpectedly deleted from alloptions cache', $option ) );
+		}
+	}
+
+	/**
+	 * Tests setting options' autoload to either 'yes' or 'no' where for some options this is already the case.
+	 *
+	 * The test also covers one option that is entirely missing.
+	 *
+	 * @ticket 58964
+	 */
+	public function test_wp_set_option_autoload_values_mixed_partial_update() {
+		global $wpdb;
+
+		$options = array(
+			'test_option1' => 'yes',
+			'test_option2' => 'no',
+			'test_option3' => 'yes',
+			'missing_opt'  => 'yes',
+		);
+		add_option( 'test_option1', 'value1', '', 'no' );
+		add_option( 'test_option2', 'value2', '', 'yes' );
+		add_option( 'test_option3', 'value3', '', 'yes' );
+		$expected = array(
+			'test_option1' => true,
+			'test_option2' => true,
+			'test_option3' => false,
+			'missing_opt'  => false,
+		);
+
+		$num_queries = get_num_queries();
+		$this->assertSame( $expected, wp_set_option_autoload_values( $options ), 'Function produced unexpected result' );
+		$this->assertSame( $num_queries + 3, get_num_queries(), 'Function made unexpected amount of database queries' );
+		$this->assertEqualSets( array( 'yes', 'no', 'yes' ), $wpdb->get_col( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name IN (" . implode( ',', array_fill( 0, count( $options ), '%s' ) ) . ')', ...array_keys( $options ) ) ), 'Option autoload values not updated in database' );
```

Comment:
We should be able to use `assertSameSets()` here instead, or did this produce failures in earlier test runs?